### ISSUE TITLE
sass fails with standard config when comments are used

### ISSUE THREAD

**Issue by steebchen** [stylelint/stylelint#3244]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Before I describe the issue here's some background information.
I am using sass files and wanted to extend from stylelint-config-standard, which unfortunately doesn't work with sass out of the box:

![selection_339](https://user-images.githubusercontent.com/5013932/38053654-ff3a0ea8-32d4-11e8-8784-4869260c9635.jpg)

my config:
```
{
	"extends": "stylelint-config-standard"
}
```

Attention! Starting from now it will get a bit hacky. If this is not how it should be done, ignore the rest. I would instead ask for a better way to use the standard stylelint config but for sass files, after a lot of digging I could not found any.

I figured I could just disable the rules which complain about the curly braces, which results in this config:
```
{
	"extends": "stylelint-config-standard",
	"rules": {
		"indentation": "tab",
		"block-opening-brace-space-before": null,
		"declaration-block-trailing-semicolon": null,
		"block-closing-brace-newline-before": null,
		"no-eol-whitespace": null
	}
}
```

It works, and stylelint correctly complains about other rules.

Now, the actual issue occurrs.

When using the sass syntax with comments, stylelint throws the following error:
```
TypeError: Cannot read property 'type' of undefined
    at indentationLevel (/home/luca/shark/account/node_modules/stylelint/lib/rules/indentation/index.js:152:22)
    at root.walk.node (/home/luca/shark/account/node_modules/stylelint/lib/rules/indentation/index.js:54:25)
    at /home/luca/shark/account/node_modules/postcss/lib/container.js:144:26
    at Root.each (/home/luca/shark/account/node_modules/postcss/lib/container.js:110:22)
    at Root.walk (/home/luca/shark/account/node_modules/postcss/lib/container.js:143:21)
    at /home/luca/shark/account/node_modules/stylelint/lib/rules/indentation/index.js:53:10
    at Promise.resolve.then (/home/luca/shark/account/node_modules/stylelint/lib/lintSource.js:221:9)
    at <anonymous>
```

I get that this is bit hacky but apart from the comments it works... 

> Which rule, if any, is this issue related to?
–

> What CSS is needed to reproduce this issue?

e.g.

```css
// this is a comment which results in an error!
.foo
  color: pink
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
	"extends": "stylelint-config-standard",
	"rules": {
		"indentation": "tab",
		"block-opening-brace-space-before": null,
		"declaration-block-trailing-semicolon": null,
		"block-closing-brace-newline-before": null,
		"no-eol-whitespace": null
	}
}
```

> Which version of stylelint are you using?

`9.1.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `yarn stylelint test.sass`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

yes, sass

> What did you expect to happen?

Not error to occur when using comments

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
TypeError: Cannot read property 'type' of undefined
    at indentationLevel (/home/luca/shark/account/node_modules/stylelint/lib/rules/indentation/index.js:152:22)
    at root.walk.node (/home/luca/shark/account/node_modules/stylelint/lib/rules/indentation/index.js:54:25)
    at /home/luca/shark/account/node_modules/postcss/lib/container.js:144:26
    at Root.each (/home/luca/shark/account/node_modules/postcss/lib/container.js:110:22)
    at Root.walk (/home/luca/shark/account/node_modules/postcss/lib/container.js:143:21)
    at /home/luca/shark/account/node_modules/stylelint/lib/rules/indentation/index.js:53:10
    at Promise.resolve.then (/home/luca/shark/account/node_modules/stylelint/lib/lintSource.js:221:9)
    at <anonymous>
```

**Comment by jeddy3** [comment]
@steebchen Thanks for the report and for using the template.

> I figured I could just disable the rules which complain about the curly braces

That's correct. This is [documented in the config README](https://github.com/stylelint/stylelint-config-standard#using-the-config-with-sugarss-syntax) (for SugarSS syntax, a whitespace-based syntax like Sass). Feel free to contribute a PR to that repo that updates the documentation to include the Sass caveats.

> When using the sass syntax with comments, stylelint throws the following error:

These type of errors are usually caused by built-in rules being incompatible with a non-standard syntax. As Sass support is experimental, it would be great if you could help narrow down the rule(s) causing the error and [contribute a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule). I don't believe any of the core team use Sass and so we rely on the help of external contributors to improve its support within stylelint.

**Comment by johndrwood** [comment]
Was this ever resolved? I'm having the same issue with comments in Sass files.

**Comment by Grawl** [comment]
but now I cannot [Turn rules off from within `.sass`](https://stylelint.io/user-guide/configuration/#turning-rules-off-from-within-your-css)

nor like this

```sass
input[type='number']
	-moz-appearance: textfield !important /* stylelint-disable-line declaration-no-important */
```

or this

```sass
input[type='number']
	-moz-appearance: textfield !important // stylelint-disable-line declaration-no-important
```

or this

```sass
input[type='number']
	/* stylelint-disable-next-line declaration-no-important */
	-moz-appearance: textfield !important
```

or this

```sass
input[type='number']
	// stylelint-disable-next-line declaration-no-important
	-moz-appearance: textfield !important
```

because of this error:

```
TypeError: Cannot read property 'start' of undefined
```

**this works**:

```sass
input[type='number']
	/*
		stylelint-disable-next-line declaration-no-important
	-moz-appearance: textfield !important
```

but I got error on this part

```
	/*
		stylelint-disable-next-line declaration-no-important
```

saying

> Please check validity of the block starting from line #19 (CssSyntaxError)

**Comment by Grawl** [comment]
https://github.com/AleshaOleg/postcss-sass/issues/72
